packages/backend/src/logger.ts

```diff
@@ -0,0 +1,69 @@
+import winston from 'winston';
+
+const levels = {
+    error: 0,
+    warn: 1,
+    info: 2,
+    http: 3,
+    debug: 4,
+};
+
+const level = () => {
+    if (process.env.LIGHTDASH_LOG_LEVEL) {
+        return process.env.LIGHTDASH_LOG_LEVEL.toLowerCase();
+    }
+    const env = process.env.NODE_ENV || 'development';
+    const isDevelopment = env === 'development';
+    return isDevelopment ? 'debug' : 'warn';
+};
+
+const colors = {
+    error: 'red',
+    warn: 'yellow',
+    info: 'green',
+    http: 'magenta',
+    debug: 'white',
+};
+
+winston.addColors(colors);
+
+const terminalFormat = winston.format.combine(
+    winston.format.timestamp({ format: 'YYYY-MM-DD HH:mm:ss' }),
+    winston.format.colorize({ all: true }),
+    winston.format.printf(
+        (info) =>
+            `${info.timestamp} [Lightdash] ${info.level}: ${info.message}`,
+    ),
+);
+
+const jsonFormat = winston.format.combine(
+    winston.format.timestamp(),
+    winston.format.json(),
+);
+
+const transports = [
+    new winston.transports.Console({
```

Comment:
With the console format, could we have a flag that toggles between `terminalFormat` vs. `jsonFormat`. GCP will index logs if they're in json format for Lightdash Cloud. But for local instances it's much more readable to have them in `terminalFormat`

https://cloud.google.com/logging/docs/structured-logging

Reply:
Ignore the above -> looks like we might need a more bespoke setup for that: https://cloud.google.com/logging/docs/setup/nodejs#using_winston

Reply:
Ah nice, gcp has his own transport. Adding it now 👍